### PR TITLE
adding add_fclcm

### PR DESCRIPTION
**Pull request by mhkhan27**

### CODE REVIEW

**Comment by mhkhan27** [comment]
[Add_fclcm_phase](https://trello.com/c/tsbjXi3M/178-addfclcmphase)

File: R/add_fclcm_phase.R [hunk 1]
```diff
@@ -1,30 +1,105 @@
...
+
+  if(!fc_phase_var %in% names(dataset)){
+    stop(glue::glue({{fc_phase_var}}, " column is not found in the dataset"))
+  }
+
+  if(!lcs_cat_var %in% names(dataset)){
+    stop(glue::glue({{lcs_cat_var}}, " column is not found in the dataset"))
+  }
+
+
```
Comment (yannsay-impact) [inline R/add_fclcm_phase.R:57]: from below the error/warnings are a bit confusing as they say "The above value(s) are not specified/possibile wrong entry" there is the print, then the table, then the error. if you have time, if we can have them glue as the messages above that would help.
(test in line 16)
image

File: R/add_fclcm_phase.R [hunk 2]
```diff
@@ -1,30 +1,105 @@
...
+  if(!all(unique(na.omit(dataset[[lcs_cat_var]])) %in% lcs_cat_all_phase)){
+    print(unique(dataset[[lcs_cat_var]])[!unique(dataset[[lcs_cat_var]]) %in% lcs_cat_all_phase])
+    stop("The above value(s) are not specified/possibile wrong entry")
```
Comment (yannsay-impact) [inline R/add_fclcm_phase.R]: message a bit confusing. Maybe trying to be clearer? "these values are present in the dataset but are not defined in the function arguments?

File: R/add_fclcm_phase.R [hunk 3]
```diff
@@ -1,30 +1,105 @@
...
+  }
+
+  if(!all(unique(na.omit(dataset[[fc_phase_var]])) %in% fc_phase_all_phase)){
+    print(unique(dataset[[fc_phase_var]])[!unique(dataset[[fc_phase_var]]) %in% fc_phase_all_phase])
+    stop("The above value(s) are not specified/possibile wrong entry")
```
Comment (yannsay-impact) [inline R/add_fclcm_phase.R]: as above

File: R/add_fclcm_phase.R [hunk 4]
```diff
@@ -1,30 +1,105 @@
...
+  }
+
+
+  if(!all(lcs_cat_all_phase %in% dataset[[lcs_cat_var]])){
+    print(lcs_cat_all_phase[!lcs_cat_all_phase %in% dataset[[lcs_cat_var]]])
+    warning("The above perameter(s) are not found in the dataset")
```
Comment (yannsay-impact) [inline R/add_fclcm_phase.R]: also a bit confusing, maybe trying to be clearer? "these values are defined as arguments in the function  but cannot be found in the dataset"?

File: R/add_fclcm_phase.R [hunk 5]
```diff
@@ -1,30 +1,105 @@
...
+  }
+
+  if(!all(fc_phase_all_phase %in% dataset[[fc_phase_var]])){
+    print(fc_phase_all_phase[!fc_phase_all_phase %in% dataset[[fc_phase_var]]])
+    warning("The above perameter(s) are not found in the dataset")
```
Comment (yannsay-impact) [inline R/add_fclcm_phase.R]: same

**Review by yannsay-impact — CHANGES_REQUESTED** [review]

File: R/add_fclcm_phase.R [hunk 6]
```diff
@@ -1,30 +1,105 @@
...
+#' test_df |> add_fclcm_phase()
+
+add_fclcm_phase <- function(dataset,
+                            fc_phase_var = "fc_phase",
+                            fc_phase_1 = "Phase 1 FC",
+                            fc_phase_2 = "Phase 2 FC",
+                            fc_phase_3 = "Phase 3 FC",
+                            fc_phase_4 = "Phase 4 FC",
+                            fc_phase_5 = "Phase 5 FC",
+                            lcs_cat_var = "lcs_cat_yes", # need to cross check with saeaad/ yann
```
Comment (yannsay-impact) [inline R/add_fclcm_phase.R]: I believe default is lcsi_cat

**Review by yannsay-impact — APPROVED** [review]

**Pull request merged**